frontend/src/metabase/query_builder/components/DimensionList.jsx

```diff
@@ -185,17 +186,21 @@ export default class DimensionList extends Component {
     } = this.props;
     const dimension = useOriginalDimension
       ? item.dimension
-      : item.dimension.defaultDimension() || item.dimension;
+      : // XXX: 7. Important note that dimension with binning options is `dimension.defaultDimension()`
+        item.dimension.defaultDimension() || item.dimension;
     const shouldExcludeBinning =
-      !enableSubDimensions &&
-      !useOriginalDimension &&
-      dimension instanceof FieldDimension &&
-      dimension.binningStrategy();
-
-    if (
-      shouldExcludeBinning ||
-      (preventNumberSubDimensions && dimension.field().isSummable())
-    ) {
+      (!enableSubDimensions &&
```

Comment:
I rearrange this condition a bit. It feels weird that we have a flag `shouldExcludeBinning` that will be used along with another condition. So I consolidate them and add a new condition that fixes the problem here.

You can read more in the PR description, but basically, the problem in the issue originates from joining a table with a native question based on the same table. In such case, a dimension's field would get all the metadata from the base table but still preserve the dimension option from the BE that specifies binning strategy = default
<img width="436" alt="image" src="https://user-images.githubusercontent.com/1937582/230497717-9852168c-3e38-4e46-8381-0f2111498bd7.png">

This causes the [binning option to be passed along with the query](https://github.com/metabase/metabase/issues/29795#issuecomment-1497314520) and causes an error.